subworkflows/local/utils_nfcore_phaseimpute_pipeline/main.nf

```diff
@@ -148,13 +148,17 @@ workflow PIPELINE_INITIALISATION {
         if (params.panel.endsWith("csv")) {
             print("Panel file provided as input is a samplesheet")
             ch_panel = Channel.fromSamplesheet("panel")
+                        .map { meta, bcf, csi, hap, legend -> [meta, bcf, csi] }
+            ch_hap_legend = Channel.fromSamplesheet("panel")
```

Comment:
Depending if the hap and legend files are given we should preprocess them or not in the panel_prep step ?

Reply:
`--steps panelprep` & `--panel` (with panel and with hap_legend) would override hap_legend and re-calculate it from the provided panel. Otherwise it will use hap_legend directly if only `--steps impute` provided.

Reply:
We should therefore add a warning in this case.
Could you add it ?